crates/lib3h/src/transport/transport_multiplex/route.rs

```diff
@@ -0,0 +1,235 @@
+    fn handle_transport_error(&mut self, error: TransportError) -> TransportResult<()> {
+        // forward
+        self.endpoint_self
+            .publish(RequestToParent::TransportError { error });
+        Ok(())
+    }
+
+    /// private dispatcher for messages coming from our parent
+    fn handle_msg_from_parent(
+        &mut self,
+        mut msg: GhostMessage<
+            RequestToChild,
+            RequestToParent,
+            RequestToChildResponse,
+            TransportError,
+        >,
+    ) -> TransportResult<()> {
+        match msg.take_message().expect("exists") {
+            RequestToChild::Bind { spec } => self.handle_bind(msg, spec),
+            RequestToChild::SendMessage { address, payload } => {
+                self.handle_send_message(msg, address, payload)
+            }
+        }
+    }
+
+    /// private handler for Bind requests from our parent
+    fn handle_bind(
+        &mut self,
+        msg: GhostMessage<RequestToChild, RequestToParent, RequestToChildResponse, TransportError>,
+        spec: Url,
+    ) -> TransportResult<()> {
+        // forward the bind to our inner_transport
+        self.inner_transport.as_mut().request(
+            std::time::Duration::from_millis(2000),
+            RouteToInnerContext::AwaitBind(msg),
+            RequestToChild::Bind { spec },
+            Box::new(|_, context, response| {
+                let msg = {
+                    match context {
+                        RouteToInnerContext::AwaitBind(msg) => msg,
+                        _ => return Err("bad context".into()),
+                    }
+                };
+                let response = {
+                    match response {
+                        GhostCallbackData::Timeout => {
+                            msg.respond(Err("timeout".into()));
+                            return Ok(());
+                        }
+                        GhostCallbackData::Response(response) => response,
+                    }
+                };
+                msg.respond(response);
+                Ok(())
+            }),
+        );
+        Ok(())
+    }
+
+    /// private handler for SendMessage requests from our parent
+    fn handle_send_message(
+        &mut self,
+        msg: GhostMessage<RequestToChild, RequestToParent, RequestToChildResponse, TransportError>,
+        address: Url,
+        payload: Vec<u8>,
+    ) -> TransportResult<()> {
+        // forward the request to our inner_transport
+        self.inner_transport.as_mut().request(
+            std::time::Duration::from_millis(2000),
+            RouteToInnerContext::AwaitSend(msg),
+            RequestToChild::SendMessage { address, payload },
+            Box::new(|_, context, response| {
+                let msg = {
+                    match context {
+                        RouteToInnerContext::AwaitSend(msg) => msg,
+                        _ => return Err("bad context".into()),
```

Comment:
Show enum value of bad context in error.

Reply:
done